assets/js/components/Notice/Icon.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Null from '../Null';
+import CheckFill from '../../../svg/icons/check-fill.svg';
+import WarningSVG from '../../../svg/icons/warning.svg';
+import StarFill from '../../../svg/icons/star-fill.svg';
+
+export const TYPES = {
+	NEW: 'new',
+	SUCCESS: 'success',
+	WARNING: 'warning',
+	INFO: 'info',
+	ERROR: 'error',
+};
+
+const typeIconMap = {
+	[ TYPES.NEW ]: StarFill,
+	[ TYPES.SUCCESS ]: CheckFill,
+	[ TYPES.INFO ]: WarningSVG,
+	[ TYPES.WARNING ]: WarningSVG,
+	[ TYPES.ERROR ]: WarningSVG,
+};
+
+export default function Icon( { className, type } ) {
+	const IconComponent = typeIconMap[ type ] || null;
+
+	if ( ! IconComponent ) {
+		return Null;
+	}
```

Comment:
I missed this before but we shouldn't return a component type from render. The `Null` component here makes sense as component type to use as a fallback (i.e. after `||` above) see https://github.com/google/site-kit-wp/blob/e2e2ab95a1745f58880e39b3e88e21db4a181e91/assets/js/components/SettingsNotice/utils.js#L38